core/arch/opnd.h

```diff
@@ -2350,19 +2433,36 @@ DR_API
 /**
  * Assumes that \p reg is a DR_REG_ constant.
  * Returns true iff it refers to an xmm (128-bit SSE/SSE2) register
- * or a ymm (256-bit multimedia) register.
+ * or a ymm (256-bit multimedia) or a zmm (512-bit multimedia) register.
```

Comment:
I would say, add a function reg_is_xmm_only() or reg_is_xmm_ex(), deprecate this (in favor of reg_is_simd), and update all callers in DR.  Maybe don't bother to add zmm.  Could be a separate PR maybe.

Reply:
I'd like to leave the zmm related things because I want to start adding these functions so I can work on a decoder. The purpose of this PR is to add zmm related things. If you don't mind.

Adding reg_is_xmm_only, I would call it reg_is_strictly_xmm.

Reply:
Done.

Reply:
But we all hate the fact that reg_is_xmm includes ymm.  Doesn't it just add fuel to the fire by piling on zmm?

Reply:
reg_is_xmm is not 100% intuitive, but I think if we're adding reg_is_strictly_xmm, then it kind of is. If reg_is_xmm returns true for xmm and ymm, but _not_ for zmm, then it's *really* completely senseless. I think having reg_is_[xmm|ymm|zmm] together with reg_is_stricly_[xmm|ymm] maybe even reg_is_strictly_zmm from completeness, is reasonable good API IMO